version.php

```diff
@@ -24,8 +24,10 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$plugin->version   = 2019111200;        // The current plugin version (Date: YYYYMMDDXX)
+$plugin->version   = 2019121800;        // The current plugin version (Date: YYYYMMDDXX)
 $plugin->requires  = 2019110500;        // Requires this Moodle version
 $plugin->component = 'metadataextractor_tika';        // Full name of the plugin (used for diagnostics)
-
 $plugin->maturity = MATURITY_ALPHA;
+$plugin->dependencies = array(
+    'local_aws' => 2017030100
```

Comment:
Everything looks good except this, I'd make it a soft dynamic dependency. Otherwise if you install the metadata plugin but only want to use some other extractor and not tika you are forced to install aws for no reason. Same model as object.

Reply:
I have removed the hard dependency and added a soft check in the server instantiation, and skipped tests for the server if it's not installed.